### DISCUSSION TITLE
Add built-in environments i.e. msvc.lua

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#2069] [Ideas]
目前要创建虚拟环境必须自己写env.lua，但写虚拟环境这个过程很容易出错（没有详细的文档，set_toolchains容易写成set_toolchain等），xmake可以加一些内置的虚拟环境来方便上手使用，也作为用户编写自己虚拟环境的参考。

举例一些常用的虚拟环境：
```
-- msvc.lua
if is_host("windows") then
  set_toolchains("msvc")
end
```

```
-- mingw.lua
add_requires("mingw-w64")
```

```
-- python.lua
add_requires("python")
```

等等。后面两个可能有待商榷，第一个msvc.lua需求量肯定很多。内置msvc环境之后，用户想在命令行使用msvc工具无需再手动配置，直接运行
```
xrepo env -b msvc shell
```
就行了，省去了命令行使用msvc的诸多不便。

**Comment by waruqi** [comment]
需求特性，你可以提到 issues 作为 feature issues

**Comment by waruqi** [comment]
加上了，`xrepo env -l` 可以看到所有
